pkg/gatherers/clusterconfig/gather_cluster_ingress_test.go

```diff
@@ -0,0 +1,51 @@
+package clusterconfig
+
+import (
+	"context"
+	"testing"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configfake "github.com/openshift/client-go/config/clientset/versioned/fake"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/insights-operator/pkg/record"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_gatherClusterIngress(t *testing.T) {
+	tests := []struct {
+		name         string
+		config       *configv1.Ingress
+		wantRecords  []record.Record
+		wantErrCount int
+	}{
+		{
+			name:   "successful retrieval of cluster ingress",
+			config: &configv1.Ingress{ObjectMeta: metav1.ObjectMeta{Name: "cluster"}},
+			wantRecords: []record.Record{
+				{
+					Name: "config/ingress",
+					Item: record.ResourceMarshaller{
+						Resource: &configv1.Ingress{ObjectMeta: metav1.ObjectMeta{Name: "cluster"}},
+					},
+				},
+			},
+			wantErrCount: 0,
+		},
+		{
+			name:         "cluster ingress not found",
+			config:       &configv1.Ingress{},
```

Comment:
Minor improvement suggestion. We could provide some name here to make more obvious that only "cluster" Ingress is gathered, but it's just a detail.